docs/migration.rst

```diff
@@ -0,0 +1,338 @@
+    except ImportError:
+        shapely_warning = None
+
+    if shapely_warning is not None and not SHAPELY_GE_20:
+        @contextlib.contextmanager
+        def ignore_shapely2_warnings():
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=shapely_warning)
+                yield
+    else:
+        @contextlib.contextmanager
+        def ignore_shapely2_warnings():
+            yield
+
+This can then be used when creating NumPy arrays (be careful to *only* use it
+for this specific purpose, and not generally suppress those warnings)::
+
+    geoms = [...]
+    arr = np.empty(len(geoms), dtype="object")
+    with ignore_shapely2_warnings():
+        arr[:] = geoms
+
+
+Consistent creation of empty geometries
+=======================================
+
+Shapely 1.x is inconsistent in creating empty geometries between various
+creation methods. A small example for an empty Polygon geometry::
+
+    # Using an empty constructor results in a GeometryCollection
+    >>> from shapely.geometry import Polygon
+    >>> g1 = Polygon()
+    >>> type(g1)
+    <class 'shapely.geometry.polygon.Polygon'>
+    >>> g1.wkt
+    GEOMETRYCOLLECTION EMPTY
+
+    # Converting from WKT gives a correct empty polygon
+    >>> from shapely import wkt
+    >>> g2 = wkt.loads("POLYGON EMPTY")
+    >>> type(g2)
+    <class 'shapely.geometry.polygon.Polygon'>
+    >>> g2.wkt
+    POLYGON EMPTY
+
+Shapely 1.8 does not yet change this inconsistent behaviour, but starting
+with Shapely 2.0, the different methods will always consistently give an
+empty geometry object of the correct type, instead of using an empty
+GeometryCollection as "generic" empty geometry object.
+
+**How do I update my code?** Those cases that will change don't raise a
+warning, but you will need to update your code if you rely on the fact that
+empty geometry objects are of the GeometryCollection type. Use the
+``.is_empty`` attribute for robustly checking if a geometry object is an
+empty geometry.
+
+In addition, the WKB serialization methods will start supporting empty
+Points (using ``"POINT (NaN NaN)"`` to represent an empty point).
+
+
+Other deprecated functionality
+==============================
+
+There are some other various functions and methods deprecated in Shapely 1.8
+as well:
+
+- The adapters to create geometry-like proxy objects with coordinates stored
+  outside Shapely geometries are deprecated and will be removed in Shapely
+  2.0 (e.g. created using ``asShape()``). They have little to no benefit
+  compared to the normal geometry classes, as thus you can convert to your
+  data to a normal geometry object instead. Use the ``shape()`` function
+  instead to convert a GeoJSON-like dict to a Shapely geometry.
+
+- The ``empty()`` method on a geometry object is deprecated.
+
+- The ``shapely.ops.cascaded_union`` function is deprecated. Use
+  ``shapely.ops.unary_union`` instead.
```

Comment:
```suggestion
  ``shapely.ops.unary_union`` instead, which internally already uses a cascaded union operation for better performance.
```

(just to indicate that the functionality / perf. benefit is not being taken away, it is just absorbed into other functions)